Python script to generate a rules file from the RST files

frank-at-adacore opened the pull request.

gusthoff commented:
Thanks for the script! 

My only question/comment is this: is there a specific reason for storing the script together with the course contents? If not, I'd prefer moving it to this directory: `frontend/tests/`

frank-at-adacore commented:
Gustavo,

The only issue I see is ease-of-use for a customer (and also the concern
that a script in "tests" might not seem to be the right place).
If putting it there would make it easy to build the document from the top
level, then that's fine

On Fri, Mar 29, 2024 at 1:54 PM Gustavo A. Hoffmann <
***@***.***> wrote:

> Thanks for the script!
>
> My only question/comment is this: is there a specific reason for storing
> the script together with the course contents? If not, I'd prefer moving it
> to this directory: frontend/tests/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AdaCore/learn/pull/1021#issuecomment-2027548065>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANFZGHZGK4VU55ECDDFM2PDY2WMFDAVCNFSM6AAAAABFOSIWNWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMRXGU2DQMBWGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Michael Frank
Mentorship and Services Engineer, AdaCore <https://www.adacore.com/>

*tel *+1 347 258 0978
150 W. 30th Street, New York, NY 10001

gusthoff commented:
This script is based on the `.rst` files, so just downloading the script from the learn website wouldn't do much. Therefore, I'll assume people have to synchronize the `learn` repository anyway. In this case, they'll have access to that directory I mentioned (`frontend/tests`) and to this particular script. (We have other scripts in that directory as well.) 

The key-point here is that we need to add some documentation about this script. For example, mentioning it in the [README.md](https://github.com/AdaCore/learn/blob/master/README.md) file and showing a simple command line on how to use the script would be great.

gusthoff commented:
Thanks for the changes. Unfortunately, there was a small misunderstanding: the [README.md](https://github.com/AdaCore/learn/blob/master/README.md) file I was referring to was the one that we already have at the root directory of the learn repository. Would you mind moving this section that you wrote to the end of that README.md file? Thanks!

gusthoff commented:
Another small comment: it would really be helpful if you include a simple command-line example on how to use the script.

frank-at-adacore commented:
Reworked this to allow the coding standards to be able to be generated standalone if the customer so desires.
Updated the README to describe how it would be done
